core/src/main/java/io/neow3j/serialization/BinaryWriter.java

```diff
@@ -100,21 +101,25 @@ public void writeFloat(float v) throws IOException {
         writer.write(array, 0, 4);
     }
 
-    public void writeInt32(int v) throws IOException {
-        buffer.putInt(0, v);
-        writer.write(array, 0, 4);
-    }
-
     /**
-     * Writes the given long (signed 64-bit integer) to the underlying output stream in little-endian order. The
-     * long's byte representation is its two's complement.
+     * Writes the first (least-significant) 16 bits of the given int (signed 32-bit integer) to the underlying output
+     * stream in little-endian order. I.e. the byte output represents an unsigned 16-bit integer in the range [0, 2^16).
      *
-     * @param v the value.
-     * @throws IOException if an I/O exception occurs.
+     * @param v the value which needs to be in the range [0, 2^16).
+     * @throws IOException              if an I/O exception occurs.
+     * @throws IllegalArgumentException if the arguments value does not lie in the interval [0, 2^16).
      */
-    public void writeInt64(long v) throws IOException {
-        buffer.putLong(0, v);
-        writer.write(array, 0, 8);
+    public void writeUInt16(int v) throws IOException {
+        if (v < 0 || v >= (int) Math.pow(2, 16)) {
+            throw new IllegalArgumentException("Value of 16-bit unsigned integer was not in interval [0, 2^16).");
+        }
+        buffer.putInt(0, v);
+        writer.write(array, 0, 2);
+    }
+
+    public void writeInt16(short v) throws IOException {
+        buffer.putShort(0, v);
+        writer.write(array, 0, 2);
     }
```

Comment:
I just noticed that `writeInt32` lacks the same - javadoc and test. It would be great if you could replicate a similar test for `writeInt32()` and add an appropriately modified javadoc on this function as well.